### PR TITLE
Arm64 cross compilation

### PR DESCRIPTION
**Pull request by sd-ricardo**
**Description** 
This branch includes all necessary changes for cross-compiling openpilot for aarch64 (cortex a-57).
New parameter added for scons compilation named 'cc' (scons cc=1 -j8 will cross compile for arm64)
Cross-compilation steps and instruction added in .md file
Added Dockerfile for development environment and a cross-compilation script that builds and generate the installation package (tarball file)
Installation package includes a setup script for installing all dependencies.
Added documentation and scripts for setting up (and running) QEMU virtual machines with Ubuntu Server (for ARM)

**Verification** 
All changes around cross-compilation does not affect any other compilation path.
We verified the cross-compiled binaries by running openpilot (headless - no ui) into an ubuntu Server virtual machines executed using QEMU. (instructions and steps added into .md file)

Collaborators:
Gonzalo Bonigo
Sampath Vengate
Shiva Aher
Pete Brown (@petebmichigan)

### CODE REVIEW

**Comment by adeebshihadeh** [comment]
Nice work, but this is a lot of code for something that isn't generally useful.

**Comment by petebmichigan** [comment]
For any sizable team, being able to cross compile a clean build at 10x the speed and being able to verify the runtime without the need for hardware is significant. Outside that, you may, at some point, want to compile with a safety certified compiler - such as Wind River's Diab compiler. This would also be cross compiling. This is very short sighted and doesn't encourage further contributions.

**Comment by adeebshihadeh** [comment]
You're right, there's value in the cross compilation stuff here. Let's make this PR the minimal set of changes to cross compile. The rest of the build environment can live in your own repo.

File: selfdrive/boardd/boardd_setup.py [hunk 1]
```diff
@@ -1,10 +1,25 @@
...
+sysroot_args=[]
+
+if CROSS_COMPILATION:
+  os.environ['CC'] = 'aarch64-linux-gnu-gcc'
+  os.environ['CXX'] = 'aarch64-linux-gnu-g++'
+  os.environ['LDSHARED'] = 'aarch64-linux-gnu-gcc -shared'
+  os.environ['LDCXXSHARED'] = 'aarch64-linux-gnu-g++ -shared'
+  os.environ["LD_LIBRARY_PATH"] = "/usr/aarch64-linux-gnu/lib"
+  sysroot_args=['--sysroot', '/usr/aarch64-linux-gnu']
+
```
Comment (adeebshihadeh) [inline selfdrive/boardd/boardd_setup.py:17]: copy/pasting this into every cython extension setup is pretty dirty. can we make these setup files into scons builders?

File: SConstruct [hunk 2]
```diff
@@ -18,15 +18,30 @@ AddOption('--asan',
 cython_dependencies = [Value(v) for v in (sys.version, distutils.__version__, Cython.__version__)]
 Export('cython_dependencies')
 
+Help("\nType: 'scons cc=1' for cross-compilation.\n")
```
Comment (pd0wm) [inline SConstruct:21]: Maybe not use `cc` as a variable/flag, that could be confused with the `CC` build variable specifying the C compiler to use.

**Comment by adeebshihadeh** [comment]
Closing due to inactivity. Feel free to reopen after making the requested changes.

**Pull request closed without merging**